cpp/src/arrow/tensor.h

```diff
@@ -114,6 +114,34 @@ class ARROW_EXPORT Tensor {
   ARROW_DISALLOW_COPY_AND_ASSIGN(Tensor);
 };
 
+template <typename TYPE>
+class ARROW_EXPORT NumericTensor : public Tensor {
+ public:
+  using TypeClass = TYPE;
+  using value_type = typename TypeClass::c_type;
+
+  /// Constructor with no dimension names or strides, data assumed to be row-major
+  NumericTensor(const std::shared_ptr<Buffer>& data, const std::vector<int64_t>& shape);
+
+  /// Constructor with non-negative strides
+  NumericTensor(const std::shared_ptr<Buffer>& data, const std::vector<int64_t>& shape,
+                const std::vector<int64_t>& strides);
+
+  /// Constructor with strides and dimension names
```

Comment:
Strides can be negative here or should you add "non-negative" as well?

Reply:
I agree with you.  I'll also add "non-negative" also in the same comment in Tensor's definition because I copied it.